oeps/oep-0042-arch-authentication.rst

```diff
@@ -0,0 +1,254 @@
+
+API Providers: Authentication Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Django REST Framework (DRF)`_ is the standard library used by Open edX to implement REST APIs in Python. Learn more about `Authentication with Django REST Framework (DRF)`_ here.
+
+The following are all DRF Authentication classes.
+
+.. list-table::
+   :widths: 60 40
+   :header-rows: 1
+
+   * - Authentication Class
+     - Status
+   * - `JwtAuthentication (edx-drf-extensions)`_
+     - Supported
+   * - `SessionAuthentication (django-rest-framework)`_
+     - Supported
+   * - `BearerAuthentication (edx-drf-extensions)`_
+     - Deprecated, except mobile
+   * - `BasicAuthentication (django-rest-framework)`_
+     - * Status Unknown
+       * Default for Blockstore, Video Pipeline, and Discovery
+       * Should this just be valid on specific endpoints for specific purposes?
+
+Note: Our implementation of JwtAuthentication is based on a fork of `JSONWebTokenAuthentication (django-rest-framework-jwt)`_ that supports Django 2.2.
+
+.. _Django REST Framework (DRF): https://www.django-rest-framework.org/
+.. _Authentication with Django REST Framework (DRF): https://www.django-rest-framework.org/api-guide/authentication/#authentication
+.. _JwtAuthentication (edx-drf-extensions): https://github.com/edx/edx-drf-extensions/blob/4d0f4de80681e5826cfbe3041ea4cda6cff87640/edx_rest_framework_extensions/auth/jwt/authentication.py#L25
+.. _SessionAuthentication (django-rest-framework): https://www.django-rest-framework.org/api-guide/authentication/#sessionauthentication
+.. _BasicAuthentication (django-rest-framework): https://www.django-rest-framework.org/api-guide/authentication/#basicauthentication
+.. _BearerAuthentication (edx-drf-extensions): https://github.com/edx/edx-drf-extensions/blob/4d0f4de80681e5826cfbe3041ea4cda6cff87640/edx_rest_framework_extensions/auth/bearer/authentication.py#L18
+.. _JSONWebTokenAuthentication (django-rest-framework-jwt): https://pypi.org/project/drf-jwt/
+
+Authenticated API Clients
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following are supported API clients that handle authentication using the supported methods documented in this OEP.
+
+* `OAuthAPIClient (edx-rest-api-client)`_: A Python client for making authenticated server-to-server calls.
+* `@edx/frontend-platform/auth (frontend-platform)`_: A JavaScript client for making authenticated calls from a micro-frontend.
+
+.. _OAuthAPIClient (edx-rest-api-client): https://github.com/edx/edx-rest-api-client/blob/518e7291f2e90d6b9dce0f943749d59fa5c1fa42/edx_rest_api_client/client.py#L181
+.. _@edx/frontend-platform/auth (frontend-platform): https://github.com/edx/frontend-platform/blob/master/README.md
+
+OAuth Backend
+~~~~~~~~~~~~~
+
+Open edX uses `EdXOAuth2 (auth-backends)`_ to provide SSO across IDAs using OAuth2. For more general information, see `Specifying authentication backends in Django`_. This backend implementation uses the `python-social-auth library`_.
```

Comment:
Do we want to stop generating local session tokens in IDAs, and instead switch to relying on the shared JWT cookie?